fendermint/vm/interpreter/src/genesis.rs

```diff
@@ -432,6 +432,43 @@ impl<'a> GenesisBuilder<'a> {
             )
             .context("failed to create activity tracker actor")?;
 
+        // F3 Certificate Manager actor - manages F3 certificates for proof-based parent finality
+        let f3_cert_state = if let Some(f3_params) = &genesis.f3 {
+            // For subnets with F3 parameters, initialize with the provided F3 data
+            let constructor_params = fendermint_actor_f3_cert_manager::types::ConstructorParams {
+                genesis_instance_id: f3_params.genesis_instance_id,
+                genesis_power_table: f3_params.genesis_power_table.clone(),
+                genesis_certificate: f3_params.genesis_certificate.clone(),
+            };
+            fendermint_actor_f3_cert_manager::state::State::new(
+                constructor_params.genesis_instance_id,
+                constructor_params.genesis_power_table,
+                constructor_params.genesis_certificate,
+            )?
+        } else {
+            // For root chains or non-IPC subnets, create with default empty state
```

Comment:
What about treating this custom actor as optional (and not merely with zero-value state)? It will be relevant only for Filecoin-anchored subnets.